src/workloads/query/GroupLikeDistinct.yml

```diff
@@ -0,0 +1,169 @@
+  Type: Loader
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        MultipleThreadsPerCollection: true
+        CollectionCount: 1
+        DocumentCount: 5000
+        BatchSize: 1000
+        Document:
+          _id: {^Inc: {start: 0, step: 1}}
+          a: {^RandomInt: {min: 0, max: 500}}
+          b: {^RandomInt: {min: 0, max: 500}}
+          c: {^RandomInt: {min: 0, max: 500}}
+        Indexes:
+          - keys: {a: 1, b: 1, c: 1}
+
+# The following use a DISTINCT_SCAN.
+- ActorFromTemplate:
+    TemplateName: RunAggCommand
+    TemplateParameters:
+      Name: "DistinctGroup"
+      OnlyActiveInPhase: 2
+      Pipeline: [{$group: {_id: "$a"}}]
+
+- ActorFromTemplate:
+    TemplateName: RunAggCommand
+    TemplateParameters:
+      Name: "DistinctGroupSort"
+      OnlyActiveInPhase: 3
+      Pipeline: [{$sort: {a: 1}}, {$group: {_id: "$a"}}]
+
+- ActorFromTemplate:
+    TemplateName: RunAggCommand
+    TemplateParameters:
+      Name: "DistinctGroupSortFirst"
+      OnlyActiveInPhase: 4
+      Pipeline: [{$sort: {a: 1, b: 1}}, {$group: {_id: "$a", accum: {$first: "$b"}}}]
+
+- ActorFromTemplate:
+    TemplateName: RunAggCommand
+    TemplateParameters:
+      Name: "DistinctGroupSortFirstFullDocument"
+      OnlyActiveInPhase: 5
+      Pipeline: [{$sort: {a: 1, b: 1, c: 1}}, {$group: {_id: "$a", accum: {$first: "$$ROOT"}}}]
+
+- ActorFromTemplate:
+    TemplateName: RunAggCommand
+    TemplateParameters:
+      Name: "DistinctGroupSortLast"
+      OnlyActiveInPhase: 6
+      Pipeline: [{$sort: {a: -1, b: -1}}, {$group: {_id: "$a", accum: {$last: "$b"}}}]
+
+- ActorFromTemplate:
+    TemplateName: RunAggCommand
+    TemplateParameters:
+      Name: "DistinctGroupSortLastFullDocument"
+      OnlyActiveInPhase: 7
+      Pipeline: [{$sort: {a: 1, b: 1, c: 1}}, {$group: {_id: "$a", accum: {$last: "$$ROOT"}}}]
+
+# The following do not yet use DISTINCT_SCAN, but they will after SERVER-84347.
+- ActorFromTemplate:
+    TemplateName: RunAggCommand
+    TemplateParameters:
+      Name: "GroupTop"
+      OnlyActiveInPhase: 8
+      Pipeline: [{$group: {_id: "$a", accum: {$top: {output: ["$b"], sortBy: {b: 1}}}}}]
+
+- ActorFromTemplate:
+    TemplateName: RunAggCommand
+    TemplateParameters:
+      Name: "GroupBottom"
+      OnlyActiveInPhase: 9
+      Pipeline: [{$group: {_id: "$a", accum: {$bottom: {output: ["$b"], sortBy: {b: 1}}}}}]
+
+- ActorFromTemplate:
+    TemplateName: RunAggCommand
+    TemplateParameters:
+      Name: "GroupTopN"
+      OnlyActiveInPhase: 10
+      Pipeline: [{$group: {_id: "$a", accum: {$topN: {n: 1, output: ["$b"], sortBy: {b: 1}}}}}]
+
+- ActorFromTemplate:
+    TemplateName: RunAggCommand
+    TemplateParameters:
+      Name: "GroupBottomN"
+      OnlyActiveInPhase: 11
+      Pipeline: [{$group: {_id: "$a", accum: {$bottomN: {n: 1, output: ["$b"], sortBy: {b: 1}}}}}]
+
+AutoRun:
```

Comment:
I wasn't sure which variants we would like for this to auto-run. Let me know if to add / remove any!